libcontainer/intelrdt/intelrdt.go

```diff
@@ -715,25 +786,47 @@ func (m *intelRdtManager) Set(container *configs.Config) error {
 	// "MB:0=5000;1=7000" which means 5000 MBps memory bandwidth limit on
 	// socket 0 and 7000 MBps memory bandwidth limit on socket 1.
 	if container.IntelRdt != nil {
-		path := m.GetPath()
+		if m.groupType != MonitoringGroupType && m.groupType != ControlGroupType {
+			return fmt.Errorf("couldn't set IntelRdt configuration: %v", NotSupportedGroupType(m.groupType))
+		}
+
 		l3CacheSchema := container.IntelRdt.L3CacheSchema
 		memBwSchema := container.IntelRdt.MemBwSchema
 
-		// Write a single joint schema string to schemata file
+		if l3CacheSchema != "" || memBwSchema != "" {
+			// Schema can be set only in Control Group.
+			if m.groupType == MonitoringGroupType {
+				if err := m.switchGroupType(m, GetIntelRdtPath); err != nil {
+					return err
+				}
+			}
+		} else {
+			// If schemata fields are empty, there is no need to have Control Group.
+			if m.groupType == ControlGroupType {
+				if err := m.switchGroupType(m, GetIntelRdtPath); err != nil {
+					return err
+				}
+				return nil
+			}
+		}
+
+		path := m.GetPath()
+
+		// Write a single joint schemata string to schemata file
 		if l3CacheSchema != "" && memBwSchema != "" {
 			if err := writeFile(path, "schemata", l3CacheSchema+"\n"+memBwSchema); err != nil {
 				return NewLastCmdError(err)
 			}
 		}
 
-		// Write only L3 cache schema string to schemata file
+		// Write only L3 cache schemata string to schemata file
```

Comment:
In my opinion, this change is unnecessary.
“schemata” is the plural of "schema". Intel RDT kernel support to write part of schemata file. We use "schema" here to indicate either L3 cache or memory bandwidth allocation strategy, and the same in config file: "l3CacheSchema", "memBwSchema".

Reply:
Automatic refactoring did this. I forgot about it.